sdk/resourcemanager/Azure.ResourceManager.Core/src/Resources/ResourceType.cs

```diff
@@ -233,65 +230,21 @@ private void Parse(string resourceIdOrType)
                 Namespace = "Microsoft.Resources";
             }
 
-            // Handle resource identifiers from RPs (they have the /providers path segment)
-            if (parts.Contains(KnownKeys.ProviderNamespace) && !KnownKeys.ProviderNamespace.Equals(parts[0], StringComparison.InvariantCultureIgnoreCase))
-            {
-                // it is a resource id from a provider
-                var index = parts.LastIndexOf(KnownKeys.ProviderNamespace);
-                for (var i = index; i >= 0; --i)
-                {
-                    parts.RemoveAt(i);
-                }
-
-                if (parts.Count < 3)
-                    throw new ArgumentOutOfRangeException(nameof(resourceIdOrType), "Invalid resource id.");
-
-                var type = new List<string>();
-                for (var i = 1; i < parts.Count; i += 2)
-                {
-                    type.Add(parts[i]);
-                }
-
-                Namespace = parts[0];
-                Type = string.Join("/", type);
-            }
-            else if (KnownKeys.ProviderNamespace.Equals(parts[0], StringComparison.InvariantCultureIgnoreCase))
-            {
-                if (parts.Count < 2)
-                {
-                    throw new ArgumentOutOfRangeException(nameof(resourceIdOrType));
-                }
-                if (!parts[1].Contains('.'))
-                {
-                    throw new ArgumentOutOfRangeException(nameof(resourceIdOrType));
-                }
-                Namespace = parts[1];
-
-                Type = string.Join("/", parts.Skip(2).Take(parts.Count - 3));
-            }
-            // Handle resource types (Micsrsoft.Compute/virtualMachines, Microsoft.Network/virtualNetworks/subnets)
+            // Handle resource types (Microsoft.Compute/virtualMachines, Microsoft.Network/virtualNetworks/subnets)
+            // Type
             else if (parts[0].Contains('.'))
             {
                 // it is a full type name
                 Namespace = parts[0];
                 Type = string.Join("/", parts.Skip(1).Take(parts.Count - 1));
             }
 
-            // Handle built-in resource ids (e.g. /subscriptions/{sub}, /subscriptions/{sub}/resourceGroups/{rg})
```

Comment:
I don't understand how we can take out all these parts without replacement.

Reply:
Feels like an area where we need some test coverage.

Reply:
Its replaced by creating a ResourceIdentifier and using the .ResourceType property of that instead of duplicating the logic here.  There were some bugs we found in how this logic was parsing child resource types and while attempting to fix it we realized we were making the same changes we already made in ResourceIdentifier.

+1 to the test coverage for both cases to validate this is no harm to the old cases and works for the cases we found that had bugs